src/main/java/uk/gov/ons/ctp/response/collection/exercise/service/impl/actionrule/AbstractActionRuleUpdater.java

```diff
@@ -0,0 +1,58 @@
+package uk.gov.ons.ctp.response.collection.exercise.service.impl.actionrule;
+
+import java.util.List;
+import java.util.UUID;
+import java.util.stream.Collectors;
+import uk.gov.ons.ctp.common.error.CTPException;
+import uk.gov.ons.ctp.response.action.representation.ActionRuleDTO;
+import uk.gov.ons.ctp.response.collection.exercise.client.ActionSvcClient;
+
+public abstract class AbstractActionRuleUpdater implements ActionRuleUpdater {
```

Comment:
Why inheritance over composition?

Reply:
No real reason tbh. Actually one of these methods is being moved out into another service class anyway in the casetype overrides to action Plan selectors refactor. I'll take filterActionRulesByType() out into a method class and remove this abstract class in that PR.